vulnerabilities/package_managers.py

```diff
@@ -489,3 +500,93 @@ async def fetch(self, pkg, session):
             pass
 
         self.cache[pkg] = versions
+
+
+class GoproxyVersionAPI(VersionAPI):
+
+    package_type = "golang"
+    module_name_by_package_name = {}
+
+    @staticmethod
+    def trim_url_path(url_path: str) -> Optional[str]:
+        """
+        github advisories for golang is using package names(e.g.
+        https://github.com/advisories/GHSA-jp4j-47f9-2vc3),
+        yet goproxy works with module names(see
+        https://golang.org/ref/mod#goproxy-protocol).
+        this method removes the last part of a package path, and returns the
+        remaining as the module name. For example: trim_url_path(
+        "https://github.com/xx/a/b") returns "https://github.com/xx/a"
+        """
+        # some advisories contains this prefix in package name, e.g. https://github.com/advisories/GHSA-7h6j-2268-fhcm
+        if url_path.startswith("https://pkg.go.dev/"):
+            url_path = url_path.removeprefix("https://pkg.go.dev/")
+        parts = url_path.split("/")
+        if len(parts) >= 2:
+            return "/".join(parts[:-1])
+        else:
+            return None
+
+    @staticmethod
+    def escape_path(path: str) -> str:
+        """
+        escape uppercase in module/version name.
+        """
+        escaped_path = ""
+        for c in path:
+            if c >= "A" and c <= "Z":
+                # replace uppercase with !lowercase
+                escaped_path += "!" + chr(ord(c) + ord("a") - ord("A"))
+            else:
+                escaped_path += c
+        return escaped_path
+
+    @staticmethod
+    async def parse_version_info(
+        version_info: str, escaped_pkg: str, session: ClientSession
+    ) -> Optional[Version]:
+        v = version_info.split()
+        if len(v) > 0:
```

Comment:
Typically you want to bail out early on condition that are not met. What about using this instead:
```suggestion
        if not v:
            return
```
then you can dedent the block below. This becomes IMHO easier to read